### DISCUSSION TITLE
Dwin display suport

### DISCUSSION THREAD

**Discussion by botioni** [arendst/Tasmota#17023] [Q&A]
Can tasmota support dwin display? Is ther any driver compiled for tasmota? 
Can it be done ?

**Comment by Jason2866** [comment]
What is dwin display? Link to a description.

**Comment by botioni** [comment]
https://sebastien.andrivet.com/en/posts/dwin-mini-dgus-display-development-guide-non-official/

https://www.dwin-global.com/tool-page/

https://how2electronics.com/interfacing-dwin-hmi-tft-lcd-display-with-arduino/

**Comment by botioni** [comment]
I have one termostat tuya with this display it is like nspanel but witout the buttons I can send pictures .  And it js a similar aproche off disigne with sdcard flashing images

**Reply by barbudor** [reply]
This is not supported by Tasmota as of today. It needs someone interrested (and having the device) willing to write the driver

If this is part of a Tuya device, are you sure that the display is connected to the ESP, and not to an TuyaMCU ?
What is the Tuya  Device ? What is the ESP on it ?

**Comment by botioni** [comment]
No tuya mcu on it it is just the display and tuya wifi it is not esp it is
an rtl but thets not a problem I swap it to esp . It is a termostat . With
color display ewrything is connected to the display sensor buzer rtl only
connects it to the internet and the relay . The display is configurable
true the software what to do like nspanel from sonoff but other
manufacturer .

This is manufacturers model with documentation.
https://www.dwin-global.com/4-inch-thermostat-hmi-touch-panelmodel-tc040c11-uw-04-product/


This is the exact model
https://m.aliexpress.com/item/1005003022387767.html?gatewayAdapt=Pc2Msite

**Comment by botioni** [comment]
I can test and help but to much to write it not thet good in c++ .
O and it is serial communication betwin the lcd and esp.

**Comment by barbudor** [comment]
There are ESP32c3 modules that are pin compatible with esp8266 such as ESP12K
ESP32c3 supports Berry language which is similar to python

**Comment by srvaddadi** [comment]
I use esp8266 as a serial bridge to send DWIN commands.
